apis/duck/v1/knative_reference.go

```diff
@@ -54,8 +64,25 @@ func (kr *KReference) Validate(ctx context.Context) *apis.FieldError {
 	if kr.Name == "" {
 		errs = errs.Also(apis.ErrMissingField("name"))
 	}
-	if kr.APIVersion == "" {
-		errs = errs.Also(apis.ErrMissingField("apiVersion"))
+	if isKReferenceGroupAllowed(ctx) {
+		if kr.APIVersion == "" && kr.Group == "" {
+			errs = errs.Also(apis.ErrMissingField("apiVersion")).
+				Also(apis.ErrMissingField("group"))
+		}
+		if kr.APIVersion != "" && kr.Group != "" {
+			errs = errs.Also(&apis.FieldError{
```

Comment:
If they do not contradict each other, should it be an error? Seems like it would be ok to update refs to add group, then drop api in two passes if the operator is feeling nervous.

Reply:
Check the last commit, now both are allowed in case they don't contradict each other